share/p2p/shrexeds/client.go

```diff
@@ -106,11 +108,15 @@ func (c *Client) doRequest(
 
 	// read and parse status from peer
 	resp := new(pb.EDSResponse)
+	err = stream.SetReadDeadline(time.Now().Add(c.params.ServerReadTimeout))
+	if err != nil {
+		return nil, fmt.Errorf("failed to set read deadline for reading status: %w", err)
+	}
```

Comment:
No need to return, we still can do request

Reply:
dont know how i didnt catch this, thanks